solutions/usaco-667.mdx

```diff
@@ -69,21 +154,110 @@ int main() {
 	ifstream read("citystate.in");
 	int N;
 	read >> N;
-	long long total = 0;
+
+	vector<pair<string, string>> pairs;
 	for (int i = 0; i < N; i++) {
 		string city;
 		string state;
 		read >> city >> state;
-		city = city.substr(0,2);
-		if (city != state) {
-			total += seen[index(state + city)];
+		city = city.substr(0, 2);  // We only care about the first two letters of the city
+		pairs.push_back({city, state});
+	}
+
+	vector<int> seen(26 * 26 * 26 * 26);
```

Comment:
```suggestion
	vector<int> seen(MAXN);
```
Make ```MAXN``` a global variable and set it equal to `26 * 26 * 26 * 26` or `pow(26, 4)`. Generally, it's a good idea to put constants as global variables...